File: Beecrowd/1. Beginner/1094/1094.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int N;
    cin >> N;

    int total = 0, rabbits = 0, rats = 0, frogs = 0;

    for (int i = 0; i < N; i++) {
        int X;
        char C;
        cin >> X >> C;

        total += X;

        if (C == 'C') rabbits += X;
        else if (C == 'R') rats += X;
        else if (C == 'S') frogs += X;
    }

    cout << "Total: " << total << " cobaias" << endl;
    cout << "Total de coelhos: " << rabbits << endl;
    cout << "Total de ratos: " << rats << endl;
    cout << "Total de sapos: " << frogs << endl;

    cout << fixed << setprecision(2);
    cout << "Percentual de coelhos: " << (rabbits * 100.0 / total) << " %" << endl;
    cout << "Percentual de ratos: " << (rats * 100.0 / total) << " %" << endl;
    cout << "Percentual de sapos: " << (frogs * 100.0 / total) << " %" << endl;

    return 0;
}

File: Beecrowd/1. Beginner/1094/1094.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);

        int n = input.nextInt();
        int total = 0, rabbits = 0, rats = 0, frogs = 0;

        for (int i = 0; i < n; i++) {
            int x = input.nextInt();
            char c = input.next().charAt(0);

            total += x;

            if (c == 'C') rabbits += x;
            else if (c == 'R') rats += x;
            else if (c == 'S') frogs += x;
        }

        System.out.println("Total: " + total + " cobaias");
        System.out.println("Total de coelhos: " + rabbits);
        System.out.println("Total de ratos: " + rats);
        System.out.println("Total de sapos: " + frogs);

        System.out.printf("Percentual de coelhos: %.2f %%\n", (rabbits * 100.0) / total);
        System.out.printf("Percentual de ratos: %.2f %%\n", (rats * 100.0) / total);
        System.out.printf("Percentual de sapos: %.2f %%\n", (frogs * 100.0) / total);

        input.close();
    }
}
